File: sophix/src/main/java/com/ysy/sophix/OtherActivity.java
package com.ysy.sophix;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.TextView;

public class OtherActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_other);

        ((TextView) findViewById(R.id.btn_other)).setText(getNB());
    }

    private String getNB() {
        if (OtherActivity.class.getSimpleName().contains("Other")) {
            return "NB";
        }
        return "BN";
    }
}

File: sophix/src/main/java/com/ysy/sophix/OtherActivity.kt
package com.ysy.sophix

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView

class OtherActivity : AppCompatActivity() {

    private val nb: String
        get() = if (OtherActivity::class.java.simpleName.contains("Other")) {
            "NB"
        } else "BN"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_other)

        (findViewById<View>(R.id.btn_other) as TextView).text = nb
    }
}
